R/weight.R

```diff
@@ -20,7 +20,7 @@
 #computing weights for inside the weighted correlation and outside penalty function
 weight <- function(t, lag, C){
   #checking if all the variables are right
-  stopifnot(length(t)/4 >= lag, is.numeric(C), all(is.numeric(t)), is.numeric(lag))
+  stopifnot(length(t)/4 >= lag, is.numeric(C), C > 0, all(is.numeric(t)), is.numeric(lag))
```

Comment:
We should add a `expect_error` test to `test_weight.R` for `C <= 0`.

Reply:
I will add this right away.